geode-core/src/main/java/org/apache/geode/internal/net/SocketCloser.java

```diff
@@ -163,60 +164,72 @@ private Future asyncExecute(String address, Runnable runnableToExecute) {
   }
 
   /**
-   * Closes the specified socket in a background thread. In some cases we see close hang (see bug
-   * 33665). Depending on how the SocketCloser is configured (see ASYNC_CLOSE_WAIT_MILLISECONDS)
+   * Closes the specified socket in a background thread. In some cases we see close hang.
+   * Depending on how the SocketCloser is configured (see ASYNC_CLOSE_WAIT_MILLISECONDS)
    * this method may block for a certain amount of time. If it is called after the SocketCloser is
    * closed then a normal synchronous close is done.
    *
    * @param socket the socket to close
    * @param address identifies who the socket is connected to
-   * @param extra an optional Runnable with stuff to execute before the socket is closed
+   * @param runBeforeClose a Runnable with stuff to execute before the socket is closed
    */
-  public void asyncClose(final Socket socket, final String address, final Runnable extra) {
-    if (socket == null || socket.isClosed()) {
+  @NotNull
```

Comment:
I don't think it makes sense to add the `@NotNull` annotation to a `void` method, since the method isn't returning anything at all.

Reply:
done